### PR TITLE
Dynamic hardware detection

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
This PR enhances the hardware parsing capabilities of keras-remote, allowing users to specify generic hardware requests (e.g., gpu-16, tpu-512) and automatically provisioning the most appropriate accelerator. It removes the strict requirement for users to know exact GKE hardware topologies, improving the overall developer experience while maintaining strict backward compatibility.

- Dynamic GPU & TPU Fallback: Added logic to dynamically query the hardware registry for generic strings (e.g., matching gpu-N and tpu-N).
- Generation-Aware Prioritization: Overhauled the fallback search to iterate through a predefined list of preferred hardware (_PREFERRED_GPUS and _PREFERRED_TPUS). This ensures requests like tpu-512 automatically provision the newest available generation (e.g. v4 or v5p over v2), rather than failing back to deprecated hardware based on dictionary insertion order.
- Canonical Accelerator Aliases: Added native regex alias support for names like v5e and ghostlite, mapping them cleanly back to the canonical v5litepod representation under the hood. This ensures backend Kubernetes node pools are labeled consistently (tpu-v5litepod-xxxx) instead of creating fragmented node pool topologies.
- Topology & Regex Hardening: Cleaned up overlapping regex matches (e.g., _MULTI_GPU_RE matching TPU strings) to improve parsing efficiency and prevent edge-case false positives. All associated unit tests in accelerators_test.py have been updated to assert proper canonical alias resolution.

### CODE REVIEW

File: keras_remote/core/accelerators.py [hunk 1]
```diff
@@ -88,6 +90,10 @@ class TpuSpec:
       4: TpuTopologySpec("2x2", "ct2-hightpu-4t", 1),
       16: TpuTopologySpec("4x4", "ct2-hightpu-4t", 4),
       32: TpuTopologySpec("4x8", "ct2-hightpu-4t", 8),
+      64: TpuTopologySpec("8x8", "ct2-hightpu-4t", 16),
+      128: TpuTopologySpec("8x16", "ct2-hightpu-4t", 32),
+      256: TpuTopologySpec("16x16", "ct2-hightpu-4t", 64),
+      512: TpuTopologySpec("16x32", "ct2-hightpu-4t", 128),
     },
   ),
```
[the remark this answers is not in the document]
Comment (divyashreepathihalli) [reply keras_remote/core/accelerators.py]: done
